### ISSUE TITLE
Windows builds fail too often due to problems with the download of Node

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#31835]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

there are multiple failures during last few days

```
Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.15.0:install-node-and-pnpm (default) on project keycloak-js-adapter-jar: Could not download Node.js from: https://nodejs.org/dist/v20.13.0/win-x64/node.exe: Could not download https://nodejs.org/dist/v20.13.0/win-x64/node.exe: Premature end of Content-Length delimited message body (expected: 69,266,072; received: 2,007,040) -> [Help 1]
```

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

The node download should complete and it should be cached

### Actual behavior

It randomly fails, and the binary is not cached.

### How to Reproduce?

See https://github.com/keycloak/keycloak/actions/runs/10197001492/job/28209152372

**Comment by miquelsi** [comment]
See https://github.com/keycloak/keycloak/actions/runs/11069721388/attempts/1

**Comment by Pepo48** [comment]
@miquelsi I experienced this in the past, in my case it was caused by a corrupted binary.

@ahus1 I'm wondering whether we could make the download more robust by improving the original https://github.com/keycloak/keycloak/pull/32053 by checking SHA checksum, which we should be able to get from https://nodejs.org/dist/v${VERSION}/SHASUMS256.txt and then compare it with the downloaded file via `sha256sum`. We could try to download the binary again if the checksums doesn't match.

I can certainly prepare a PR in case you think that it would make the things more stable.
